### PR TITLE
CI: dependabot config for major versions

### PR DESCRIPTION
**Pull request by MattBuckley-TfN**
# Changes

> [!IMPORTANT]
> Using tomorrow's date for testing purposes, will change this to the 28th once we've confirmed it works.

Added dependabot config for major versions of Python packages and minor versions of CAF packages. Also, check on github actions.

# Tasks

- [x] Have unittests been added (testing individual functions and their edge cases) - No
- [x] Has documentation (including user guide) been updated - No
- [x] Have new dependencies been added (or changed) in relevant `requirements.txt` - No
- [ ] Code linting, tests and other workflows pass

----
📚 Documentation preview 📚: https://caftoolkit--231.org.readthedocs.build/en/231/

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+version: 2
+updates:
+  - package-ecosystem: "pip"
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "cron"
+      cronjob: "0 0 18 * *"
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply .github/dependabot.yml]: This is incorrect, see https://docs.github.com/en/code-security/reference/supply-chain-security/dependabot-options-reference#schedule-

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        update-types:
+          - "version-update:semver-minor"
+    # If you only want security update PRs, set this to 0
+    open-pull-requests-limit: 3
+
+  - package-ecosystem: "github-actions"
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "cron"
+      cronjob: "0 0 28 * *"
```
[the remark this answers is not in the document]
Comment (MattBuckley-TfN) [reply .github/dependabot.yml:24]: This is incorrect, see https://docs.github.com/en/code-security/reference/supply-chain-security/dependabot-options-reference#schedule-
